packages/rps/src/redux/game-v2/saga.ts

```diff
@@ -0,0 +1,191 @@
+import {take, select, call, put} from 'redux-saga/effects';
+// import { take } from 'redux-saga/effects';
+import {RPSChannelClient} from '../../utils/rps-channel-client';
+import {AppData, hashPreCommit, calculateResult, updateAllocation, Player} from '../../core';
+import {updateChannelState, chooseSalt, resultArrived, restart, FundingSituation} from './actions';
+import {GameState} from './state';
+import {randomHex} from '../../utils/randomHex';
+import {bigNumberify, BigNumber} from 'ethers/utils';
+
+const getGameState = (state: any): GameState => state.game;
+
+export function* gameSaga(channelClient: RPSChannelClient) {
+  while (true) {
+    yield take('*'); // run after every action
+
+    const {localState, channelState}: GameState = yield select(getGameState);
+
+    if ('player' in localState && localState.player === 'A') {
```

Comment:
It might be clearer to separate out the branching logic from the effects (as we tend to do in reducers). As in, each effect is a *single* function call with a descriptive name.

You could also `switch` on `localState.type`, then check the `channelState` inside each branch. This would make it easier to compare to the flowchart diagram.

Reply:
Yep, agreed. Planning to refactor this whole function later, to make it easier to follow.